test/src/test/java/org/corfudb/integration/ClusterReconfigIT.java

```diff
@@ -599,4 +600,138 @@ public void testSystemDownHandler() throws Exception {
     }
+
+    /**
+     * Ensure that multiple resets on the same epoch reset the log unit server only once.
+     * Consider 3 nodes 9000, 9001 and 9002.
+     * Kill node 9002 and recover it so that the heal node workflow is triggered, the node is
+     * reset and added back to the chain.
+     * We then perform a write on addresses 5-9.
+     * Now we attempt to reset the node 9002 again on the same epoch and assert that the write
+     * performed on address 1 is not erased by the new reset.
+     */
+    @Test
+    public void preventMultipleResets() throws Exception {
+        // Set up cluster of 3 nodes.
+        final int PORT_0 = 9000;
+        final int PORT_1 = 9001;
+        final int PORT_2 = 9002;
+        Process corfuServer_1 = runUnbootstrappedPersistentServer(corfuSingleNodeHost, PORT_0);
+        Process corfuServer_2 = runUnbootstrappedPersistentServer(corfuSingleNodeHost, PORT_1);
+        Process corfuServer_3 = runUnbootstrappedPersistentServer(corfuSingleNodeHost, PORT_2);
+        final Layout layout = get3NodeLayout();
+
+        final int retries = 3;
+        BootstrapUtil.bootstrap(layout, retries, PARAMETERS.TIMEOUT_SHORT);
+
+        CorfuRuntime runtime = createDefaultRuntime();
+        UUID streamId = CorfuRuntime.getStreamID("testStream");
+        IStreamView stream = runtime.getStreamsView().get(streamId);
+        byte[] payload = "test".getBytes();
+
+        // Shutdown server 3.
+        shutdownCorfuServer(corfuServer_3);
+        waitForEpochChange(refreshedEpoch -> refreshedEpoch > layout.getEpoch(), runtime);
+        final Layout layoutAfterFailure = runtime.getLayoutView().getLayout();
+
+        // Write at address 0-4.
+        stream.append(payload);
+        stream.append(payload);
+        stream.append(payload);
+        stream.append(payload);
+        stream.append(payload);
+
+        // Restart server 3 and wait for heal node workflow to modify the layout.
+        corfuServer_3 = runUnbootstrappedPersistentServer(corfuSingleNodeHost, PORT_2);
+        waitForEpochChange(refreshedEpoch -> refreshedEpoch > layoutAfterFailure.getEpoch(), runtime);
+
+        // Write at address 5-9.
+        final long startAddress = 5L;
+        final long endAddress = 9L;
+        stream.append(payload);
+        stream.append(payload);
+        stream.append(payload);
+        stream.append(payload);
+        stream.append(payload);
+
+        // Trigger a reset on the node.
+        runtime.getLayoutView().getRuntimeLayout(layoutAfterFailure)
+                .getLogUnitClient("localhost:9002")
+                .resetLogUnit(layoutAfterFailure.getEpoch()).get();
+
+        runtime.getLayoutView().getRuntimeLayout()
+                .getLogUnitClient("localhost:9002")
+                .read(Range.closed(startAddress, endAddress)).get()
+                .getAddresses().values().forEach(logData -> {
+            assertThat(logData.isEmpty()).isFalse();
```

Comment:
probably want to check the data contents. isEmpty will return false for a Hole/trimmed.